src/v/json/json.h

```diff
@@ -95,17 +94,6 @@ void rjson_serialize(
     w.EndArray();
 }
 
-template<typename T, size_t max_fragment_size>
-void rjson_serialize(
-  json::Writer<json::StringBuffer>& w,
-  const fragmented_vector<T, max_fragment_size>& v) {
-    w.StartArray();
-    for (const auto& e : v) {
-        rjson_serialize(w, e);
-    }
-    w.EndArray();
-}
-
 template<typename T, size_t chunk_size = 128>
 void rjson_serialize(
```

Comment:
Don't we need to fix these other functions too?

Reply:
By fix I mean move to the public header? The public header is re-exporting these methods right now.

Reply:
sorry i'm not sure what you mean. what's an example?

Reply:
Sorry I was confused that the new `json.h` file is in `container/include/container`, originally I thought you put that in `json/include/json`. I am assuming that moving json serialization to the container module is intentional? Should we be doing that for everything? Or maybe we should be doing the serde approach and just have a template version?

I am assuming you're moving `rjson_serialize` for fragmented_vector for some cyclical dependency?

Reply:
Yeh, definitely open to alternatives here. The idea is that it is not scalable for `json/json.h` to have overrides for everything (e.g. container/* net/*) and that each module should have its own `json.h` for its types. what remains in `json/json.h` would be overloads for external dependencies and stuff in base/*.

wdyt?

Reply:
That SGTM